rust/src/dns/dns.rs

```diff
@@ -825,6 +829,19 @@ pub extern "C" fn rs_dns_probe(input: *const u8, len: u32)
     return 0;
 }
 
+#[no_mangle]
+pub extern "C" fn rs_dns_probe_srv(input: *const u8, len: u32)
+                                   -> u8
+{
+    let slice: &[u8] = unsafe {
```

Comment:
you can use the `build_slice!` macro here.

Reply:
ok